chromebrew/chromebrew
# Split check command into separate file

**Split out `check` command to `commands/check.rb`.**

As I have split out many of these changes into separate PRs, the ones that are already in master/no longer relevant have been crossed out.

Behavior changes:
- ~~Replace `crew test` with a Rakefile and `crew check`.~~
- Rewrite the message printed when `ruby_rubocop` is not installed.
- Remove the `crew check -v` behaviour.
  - Calling `version.rb` with the name of the package works the same, and dropping this wrapper allows us to drop the `-V` parameter which was at cross purposes with the same parameter when used like so: `crew -V`.
- ~~Make `prop_test` and `buildsystem_test` less verbose by default.~~
  - ~~This also makes it a lot easier to read the unit test logs.~~
- Fix the help message for `crew check`.
- Remove `zstd` instead of `zlib` in `test_force_remove_essential_package` in `tests/commands/remove.rb`, as a missing zlib causes rubocop failures on i686.
  
Refactoring changes:
(`prop_test` and `buildsystem_test`)
- ~~Return literal '1' and '0' and add those rather than using `@tofail`.~~
- ~~Use `CREW_PACKAGES_PATH` and `CREW_LIB_PATH` rather than relative directory paths.~~
- ~~Take an actual package object in `check_properties` rather than using a name and an instance variable.~~
- ~~Use less single-use variables in and around `check_buildsystem`.~~

(`bin/crew` and `commands/check.rb`)
- Merge `check_package` and `copy_package` into a single function, using the `to_copy` variable to replicate the previous method of outsourcing both the prompting for input and subsequent copying to `copy_package`.
  - As `crew build`, `crew install` and `crew reinstall` all now call `Command.check`, we rework the case of there being no local package to check to result in a silent ignore when we are calling it from another method (so you can install packages without having to worry about being in `CREW_LOCAL_REPO_ROOT`, but we let the user know about this happening when it is a call from `crew check`.
- Use `CREW_LIB_PATH` instead of relative paths.
- Use `File.join` instead of string interpolation.

(`prop_test`, `buildsystem_test` and `dep_test`)
- Return a failure code (when tests fail) when ran against a single file.

Tested and working on `x86_64`.

### Run the following to get this pull request's changes locally for testing.
```
CREW_REPO=https://github.com/Zopolis4/chromebrew.git CREW_BRANCH=projectregula8 crew update
```
